### ISSUE TITLE
Non-cubic data preprocessing

### ISSUE THREAD

**Issue by maberrospi** [OSUPCVLab/SegFormer3D#27]
Hi,

Thanks for sharing your research and the model weights for future research!

I noticed that the model input has to have a cubic shape (e.g. 128x128x128) as was the case for the BraTs experiment. Could you please share how you can deal with non-cubic data? For example having a look at the ACDC dataset I noticed that in that case the data is not cubic. In other words the number of slices is much less than the height and width. How did you preprocess this type of data to be compatible with your model architecture?

Looking forward to your response!

**Comment by maberrospi** [comment]
Thanks for the prompt reply! That is good to hear and again thank you for your efforts! Looking forward to that release. 

You mention this will support variable-sized cubes. My question mostly relates to how you deal with non-cubic shaped inputs. For example in the ACDC dataset which you report in your paper has a smaller Depth channel compared to Height and Width. If possible I would like to know what approach you used to preprocess this data. I could think of interpolation along the Depth, padding or downscaling Height and Width but I believe all these methods have some tradeoffs especially when it comes to medical images.

Looking forward to your input!

**Comment by maberrospi** [comment]
Thank for this information! I must have missed the mention of nnFormer for data processing in the repo. Also, thank you for the further explanation! I will have a look at the nnFormer codebase.

**Comment by innat** [comment]
@bnavard 
To address this issue, I think you mainly relied on augmentation or preprocessing methods. I’ve noticed similar functionality in MONAI’s transformation pipeline, where interpolation is often used under the hood in dataloader. Instead of depending on augmentation, another approach could be to design the model architecture itself to support arbitrary input shapes. Have you explored this direction?

At the moment, I’m implementing this model in Keras 3 (supporting both 2D and 3D), similar to the official code, and plan to add it to a repo ([medic-ai](https://github.com/innat/medic-ai)) I’m building — a MONAI-like medical imaging toolkit fully written in Keras 3 with support for all backends. I haven’t tested the arbitrary input shape option yet, but I believe knowing your thoughts on it can provide some useful intuition moving forward. Thanks.
